refactor/uncrustify

dundargoc opened the pull request:
- refactor(uncrustify): disable formatting in problematic code sections
- refactor(uncrustify): format all c code under /src/nvim/
- refactor(uncrustify): change rules to better align with the style guide

dundargoc committed:
refactor(uncrustify): disable formatting in problematic code sections

dundargoc commented:
Still haven't come up with a good way to add whitespaces around `=` since it messes up the `INIT` macros.

justinmk commented on src/nvim/api/extmark.c:
```diff
@@ -559,6 +561,8 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id, Integer line, Integer
     { NULL, NULL, NULL },
   };
 
+  // uncrustify:on
```
might be clearer if this sticks next to the relevant code, i.e. no blank line. same for the start.

dundargoc committed:
refactor(uncrustify): format all c code under /src/nvim/

dundargoc committed:
refactor(uncrustify): change rules to better align with the style guide

Add space around arithmetic operators '+' and '-'.
Remove space between back-to-back parentheses, i.e. ')(' vs. ') ('.
Remove space between '((' or '))' of control statements.
Add space between ')' and '{' of control statements.
Remove space between function name and '(' on function declaration.
Collapse empty blocks between '{' and '}'.
Remove newline at the end of the file.
Remove newline between 'enum' and '{'.
Remove newline between '}' and ')' in a function invocation.
Remove newline between '}' and 'while' of 'do' statement.

bfredl commented:
This time, I will allow it.
